### ISSUE TITLE
Performance regressions in NVDA with Windows Terminal

### ISSUE THREAD

**Issue by Neurrone** [nvaccess/nvda#14189]
/CC @codeofdusk 

### Steps to reproduce:

Use Windows Terminal. In particular:

* Type in text
* Use vim to edit a text file and exit, or use git log - anything that clears the screen and switches back once the command finishes
* Use commands like git status

Basically normal use of the terminal

### Actual behavior:

I've noticed that NVDA now lags noticeably:

* When typing characters into the prompt, there is now a very noticeable delay before key echo occurs
* When the screen clears such as when exiting editing of a text file with Vim, there is again a noticeable delay before NVDA starts reading text
* When using commands like git status and nvda+up arrow to read previous lines, NVDA seems to be "stuck" on the last line or two of output for a few seconds before nvda + up arrow is able to read previous lines of output beyond the last 2 lines.

### Expected behavior:

None of this should be happening, unsure if it s aregression with NVDA or Terminal

### NVDA logs, crash dumps and other attachments:
None

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-26524,bc1f92f9 (2022.4.0.26524)

#### Windows version:
10.0.19044.2006

#### Name and version of other software in use when reproducing the issue:
Windows terminal 1.14.2281.0

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I have also seen this on another laptop which is on a more recent alpha build

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

**Comment by seanbudd** [comment]
@codeofdusk @carlos-zamora - any ideas on this one?

**Comment by Neurrone** [comment]
After a few weeks of usage, I haven't seen the first item occur often, maybe my computer was under high load or something.

I suspect that the second item happens when there is a lot of text already in the terminal although I'd need to test this more.

The third item where it gets stuck on the first couple of most recent lines for a while before you can read further above happens for me all the time and is something I can reliably reproduce.

**Comment by brunoprietog** [comment]
I can confirm this, especially when exiting vim

**Comment by seanbudd** [comment]
@brunoprietog, which version of NVDA are you using?

**Comment by brunoprietog** [comment]
NVDA alpha-27157,0fc81e9f

**Comment by tage64** [comment]
I have this annoying issue all the time, every day, and it is very frustrating. Specificly I have the "third" problem where moving the review cursor becomes slow after a massive flow of terminal output. It helps to restart the terminal and restart NVDA, but I have to do that every 30 minutes or so which is frustrating.

If anyone has any idea how to solve this issue or want to investigate this, I would be very thankful.

Best regards,
Tage

**Comment by Adriani90** [comment]
@Neurrone, @brunoprietog, @tage64 can you test please with the last alpha version of NVDA? The one which contains code from #14888? This might have fixed this issue.

**Comment by tage64** [comment]
Yes! The issue seem to be fixed in the latest alpha ([alpha-30080,dd8a44cf](https://ci.appveyor.com/project/nvaccess/nvda/build/alpha-30080,dd8a44cf))!   Thanks alot!

**Comment by Neurrone** [comment]
@tage64 what version of NVDA and Windows terminal are you using?

I haven't seen this in a while, even without the code from #14888

**Comment by Adriani90** [comment]
Thanks for the update, closing as works for me.
